bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -5340,14 +5328,12 @@ void ReplicaImp::handleDeferredRequests() {
     }
     // Currently we are avoiding duplicates on deferred RO requests queue
     while (!deferredRORequests_.empty()) {
-      auto *msg = deferredRORequests_.front();
+      auto msg = std::move(deferredRORequests_.front());
```

Comment:
not sure why we are moving twice here?

How about:
```
while (!deferredRORequests_.empty()) {
      auto& msg = deferredRORequests_.front();
      auto msgHandlerCallback = msgHandlers_->getCallback(msg->type());
      if (msgHandlerCallback) {
        msgHandlerCallback(std::unique_ptr<MessageBase>(std::move(msg)));
      }
      deferredRORequests_.pop_front();
      deferredRORequestsMetric_--;
    }
```

Reply:
done